### PR TITLE
feat(admin):  message type admin frontend

### LINKED ISSUE #3027: 🚀 [Feature] moderator comments frontend
## 🚀 Feature

- [x] based on: #3018
  - [x] extend the datamodel of contribution-message with the additional moderator-marker
- [x] extent contribution message pages in admin interface
  - [x] checkbox to mark a new message as a moderator-message
- [ ] extent contribution message page in wallet
  - [ ]  hide/filter moderator-message (should be filtered in backend)

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
work of @Mogge  PR https://github.com/gradido/gradido/pull/3072

### Issues
- fixes #3027 

Bildschirmfoto vom 2023-06-21 13-27-17

### CODE REVIEW

File: admin/src/components/ContributionMessages/ContributionMessagesFormular.vue [hunk 1]
```diff
@@ -12,8 +12,27 @@
...
+          <b-col>
+            <b-button
+              type="button"
+              variant="warning"
+              class="text-black"
+              :disabled="disabled"
+              @click="onSubmit('MODERATOR')"
+              data-test="submit-moderator"
+            >
+              Moderator Notiz
```
Comment (Mogge) [inline admin/src/components/ContributionMessages/ContributionMessagesFormular.vue:24]: Please localize. `Moderator note`

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+      <small
+        class="ml-4 text-success"
+        :class="message.type === 'MODERATOR' ? 'text-success' : 'text-warning'"
+      >
+        {{ $t('moderator') }}
+      </small>
       <parse-message v-bind="message"></parse-message>
+      <small v-if="message.type === 'MODERATOR'">
+        <hr />
+        Diese Nachricht ist nur für die Moderatoren sichtbar!
```
Comment (Mogge) [inline admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue:20]: Localize. `This message can only be seen by moderators`

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue [hunk 3]
```diff
@@ -1,11 +1,24 @@
 <template>
   <div class="contribution-messages-list-item">
-    <div v-if="message.isModerator" class="text-right is-moderator">
+    <div
+      v-if="message.isModerator"
+      class="text-right is-moderator"
+      :class="message.type === 'MODERATOR' ? 'is-hidden-moderator-message p-2' : ''"
```
Comment (Mogge) [inline admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue:6]: I cannot see any changes of the layout

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
       <span class="ml-2">{{ $d(new Date(message.createdAt), 'short') }}</span>
-      <small class="ml-4 text-success">{{ $t('moderator') }}</small>
+      <small
+        class="ml-4 text-success"
+        :class="message.type === 'MODERATOR' ? 'text-success' : 'text-warning'"
+      >
+        {{ $t('moderator') }}
+      </small>
       <parse-message v-bind="message"></parse-message>
+      <small v-if="message.type === 'MODERATOR'">
```
Comment (Mogge) [inline admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue:18]: Text is not visible

File: admin/src/graphql/adminCreateContributionMessage.js [hunk 5]
```diff
@@ -1,8 +1,12 @@
 import gql from 'graphql-tag'
 
 export const adminCreateContributionMessage = gql`
-  mutation ($contributionId: Int!, $message: String!) {
-    adminCreateContributionMessage(contributionId: $contributionId, message: $message) {
+  mutation ($contributionId: Int!, $message: String!, $messageType: ContributionMessageType) {
```
Comment (Mogge) [inline admin/src/graphql/adminCreateContributionMessage.js:4]: `mesageType` was missing here.

**Review by Mogge — CHANGES_REQUESTED** [review]
Please remove the unused query `listContributionMessages`

File: admin/src/graphql/adminCreateContributionMessage.js (hunk 5 above)
Comment (ogerly) [reply admin/src/graphql/adminCreateContributionMessage.js:4]: done

File: admin/src/components/ContributionMessages/ContributionMessagesFormular.vue (hunk 1 above)
Comment (ogerly) [reply admin/src/components/ContributionMessages/ContributionMessagesFormular.vue:24]: done

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue (hunk 2 above)
Comment (ogerly) [reply admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue:20]: done

**Comment by ogerly** [comment]
Bildschirmfoto vom 2023-06-26 10-08-45
Bildschirmfoto vom 2023-06-26 10-08-00

**Comment by ogerly** [comment]
image

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue (hunk 3 above)
Comment (ogerly) [reply admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue]: done

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue (hunk 4 above)
Comment (ogerly) [reply admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue]: done

**Comment by ogerly** [comment]
FireShot Capture 016 - Gradido Admin Interface - localhost

**Review by mahula — APPROVED** [review]

**Pull request merged**